base_multi_company/models/multi_company_abstract.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class MultiCompanyAbstract(models.AbstractModel):
+
+    _name = 'multi.company.abstract'
+    _description = 'Multi-Company Abstract'
+
+    company_id = fields.Many2one(
+        string='Company',
+        comodel_name='res.company',
+        compute='_compute_company_id',
+        store=True,
```

Comment:
Well, I haven't tried that. First impulse is to hide that field from views. Second one is to not hide, but see adding `compute_sudo=True`. The only doubt is how name_get() calls are done, which is the possible read errors that can happen. Have you tried?

Other option is to change also the comodel_name to the new SQL view, which won't have access problems at all.